docs/admin/setup/distributed.rst

```diff
@@ -504,6 +504,118 @@ It relays message between instances:
       :align: center
 
 
+Integrated Code Lifecycle
+^^^^^^^^^^^^^^^^^^^^^^^^^
+The integrated code lifecycle (ICL) can integrate build agents into a multi instance server setup. In ICL, we differentiate between two types of server node types: **core nodes** and **build agent nodes**.
+Core nodes provide the full Artemis functionality, while build agents simply execute build jobs for the testing of programming exercises.
+Both node types run the Artemis application, albeit with different profile sets and different application configuration files.
+Compared to core nodes, build agents nodes are much more light-weight, as they have less service dependencies and provide less functionality. Thus, they require less system resources and start much quicker than core nodes.
+
+The previously mentioned steps concerning the multiple Artemis instance setup remain unchanged, as we only need to adapt the run and application configurations for each node.
+
+**Core nodes**
+
+Core nodes serve the main functionality of Artemis. In ICL, this additionally includes the :ref:`CI Management <ci_management>`,
+responsible for managing and interacting with the build job queue (adding, cancelling and viewing build jobs), and the :ref:`Local VC system <local_vc>`.
+
+For ICL, the run configuration for core nodes need to include the additional profiles ``core``, ``localvc`` and ``localci``, e.g.:
+
+::
+
+        --spring.profiles.active=prod,core,ldap-only,localvc,localci,athena,scheduling,iris,lti
+
+Core nodes do not require further adjustments to the ``application-prod.yml``, as long as you have added the necessary variables as described in the :ref:`Integrated Code Lifecycle Setup <Integrated Code Lifecycle Setup>`.
+
+**Build Agents**
+
+Build agents can be added to and removed from the server cluster depending on the build capacity needed to conduct the automatic
+assessment of programming exercises. If desired, build agents can execute multiple build jobs concurrently. In this case, you need to make sure that the server node your build agents is running on has enough resources.
+We recommend at least 2 CPUs and 2 GB of RAM for each concurrently running build job.
+
+The run configuration contains just two profiles:
+
+::
+
+    --spring.profiles.active:prod,buildagent
+
+Build agents depend on much less services than the core nodes, thus we can adapt the ``application-prod.yml`` to exclude some of these dependencies.
+This heavily reduces the application start up time and resource demand.
+
+You can make following adaptations to the ``application-prod.yml``:
+
+1. Disable Liquibase and loadbalancer cache:
+
+.. code-block:: yaml
+
+    spring:
+        liquibase:
+            enabled: false
+        cloud:
+            loadbalancer:
+                cache:
+                    enabled: false
```

Comment:
```suggestion
   .. code-block:: yaml

       spring:
           liquibase:
               enabled: false
           cloud:
               loadbalancer:
                   cache:
                       enabled: false
```
The items that belong to the bullet-point should be indented accordingly. Same for 2.-4. below.